### ISSUE TITLE
How can I format my `volumes` section correctly in docker-compose.yaml?

### ISSUE THREAD

**Issue by J-Liu** [prettier/prettier#4971]
I'm using prettier as my YAML file formatter, but I got some troubles today I edit my `docker-compose.yaml`.

the `volumes` section should looks like this, and I typed it like this.
```yaml
...
volumes:
  postgres:
  redis:
```

When I saving it, my vim call prettier automatically to format the code, and I got this
```yaml
...
volumes:
  ? postgres
  ? redis
```

How can I get a `right` and `formatted` `docker-compose.yaml` file?
Or, please fix it, if it is a bug.

Thanks all.

**Comment by kachkaev** [comment]
This is a feature of yaml, not a bug. If you don’t like question marks before field names, you can add `~` after `:`, which will also indicate null:

```yaml
volumes:
  postgres: ~
  redis: ~
```

Prettier should keep things as is in this  case.

**Comment by J-Liu** [comment]
@kachkaev Thank you.
And, I'm new to docker-compose, if I add `~`, will the file still working properly?

**Comment by kachkaev** [comment]
It should as this is the same data-wise.

**Comment by J-Liu** [comment]
@ikatyang Aha! Your patch is really great! Will this patch be accept soon or already accepted? I'm not familiar with GitHub pull requests but patch set in mailing lists.
@kachkaev It should, but I still think we should obey the docker rules.

**Comment by J-Liu** [comment]
Thank you, I will install it from github, and `npm upgrade -g` after the npm released.

**Comment by J-Liu** [comment]
1.14.3 still have this problem
This is very common in docker-compose file, please fix it.
```
version: '3'

services:
  redis:
    volumes:
      - redis_data:/redis_data

  mysql:
    volumes:
      - ./mysql_data:/var/lib/mysql

volumes:
  mysql_data:
  redis_data:
```

**Comment by ikatyang** [comment]
@J-Liu As mentioned in the [changelog](https://github.com/prettier/prettier/blob/master/CHANGELOG.md#1143), 1.14.3 is just a quick patch for adding the missing LICENSE. 1.15 will probably be released after getting #5098 done.

**Comment by J-Liu** [comment]
OK, thank you. U save docker compose:)
